lessons/basics/enum.md

```diff
@@ -12,9 +12,94 @@ A set of algorithms for enumerating over collections.
 
 ## Enum
 
-The `Enum` module includes over one hundred functions for working with the collections we learned about in the last lesson.
+The `Enum` module includes nearly 100 functions for working with the collections we learned about in the last lesson.
 
-This lesson will only cover a subset of the available functions, to see a complete set of functions visit the official [`Enum`](http://elixir-lang.org/docs/stable/elixir/Enum.html) docs; for lazy enumeration use the [`Stream`](http://elixir-lang.org/docs/stable/elixir/Stream.html) module.
+This lesson will only cover a subset of the available functions, however we can actually enamine them ourselves.
+Let's do a little experiment in IEx.
+
+```elixir
+$ iex
+iex(1)> Enum.__info__(:functions) |> Enum.each(fn({function, arity}) ->
+...(1)>   IO.puts "#{function}/#{arity}"
+...(1)> end)
+all?/1
+all?/2
+any?/1
+any?/2
+at/2
+at/3
+...
+```
+We can take this even further.
+By enumerating in this manner we can even get the documentation for every function of a given module by leveraging the `h` operator, which prints docs.
+
+```elixir
+$ iex
+iex(1)> Enum.__info__(:functions) |> Enum.each(fn({function, arity}) ->
+...(1)>   Code.eval_string("require IEx.Helpers; IEx.Helpers.h Enum.#{function}")
```

Comment:
Agreed upon further thought.

Fun idea: Write a function that generates the entirety of the elixir documentation and prints it to a file or generates a PDF using pandoc with this technique iterating through all std library modules.